### PR TITLE
Add config for stackrox ui component test job

### PR DESCRIPTION
**Pull request by dvail**
Step 1 from [Middle change: Adding a new job to OpenShift CI for stackrox/stackrox](https://docs.engineering.redhat.com/pages/viewpage.action?spaceKey=StackRox&title=Making+changes+to+OpenShift+CI). 

Adds a new job that runs UI component tests. These tests will run in Cypress, like the ui-e2e tests, but are entirely self-contained and do not require an openshift cluster.

The purpose of this job is to allow a level of test abstraction for UI code between low-level unit tests and high level e2e tests.

This job:
- is currently optional
- does not automatically run
- will upload screenshots and videos to GCS, like the e2e job

Step 2 to add a supporting job in stackrox/stackrox is in progress.

### CODE REVIEW

File: ci-operator/config/stackrox/stackrox/stackrox-stackrox-master.yaml [hunk 1]
```diff
@@ -57,6 +57,13 @@ tests:
       COLLECTION_METHOD: core_bpf
     workflow: stackrox-stackrox-e2e-job
   timeout: 5h0m0s
+- always_run: false
+  as: ui-component-tests
+  optional: true
+  run_if_changed: ^ui/
```
Comment (gavin-stackrox) [inline ci-operator/config/stackrox/stackrox/stackrox-stackrox-master.yaml:63]: Can you remove this 'for now'. IIUC `always_run: false` has precedence, so this job should not run on any PRs unless explicitly requested but I'm never 100% sure what prow will do.
```suggestion
```

**Review by gavin-stackrox — CHANGES_REQUESTED** [review]

**Review by gavin-stackrox — APPROVED** [review]

**Comment by gavin-stackrox** [comment]
@dvail see https://github.com/openshift/release/pull/46991#issuecomment-1865016590 for the last step to merge this PR.

**Comment by dvail** [comment]
/pj-rehearse pull-ci-stackrox-stackrox-master-ui-component-tests

**Pull request merged**